src/block_group.rs

```diff
@@ -104,4 +109,20 @@ impl BlockGroupDescriptor {
 
         Ok(block_group_descriptor)
     }
+
+    /// Read all block group descriptors.
+    pub(crate) fn read_all(
+        sb: &Superblock,
+        reader: &mut dyn Ext4Read,
+    ) -> Result<Vec<BlockGroupDescriptor>, Ext4Error> {
+        let mut block_group_descriptors =
+            Vec::with_capacity(usize_from_u32(sb.num_block_groups));
+
+        for bgd_index in 0..sb.num_block_groups {
+            let bgd = Self::read(sb, reader, bgd_index)?;
```

Comment:
curious about using `Self` here and `BlockGroupDescriptor` elsewhere (in the return types).

Why not `Self` everywhere? Why not `BlockGroupDescriptor` everywhere?

Reply:
Unintentional divergence. Changed everything to use `Self`.